### ISSUE TITLE
GDScript: missing/possible constant folding

### ISSUE THREAD

**Issue by ThakeeNathees** [godotengine/godot#41673]
**Godot version:** 5abb53be689475cf368b1d032bd3aecbe0f18051

**OS/device including version:** windows10

**Issue description:** 
Most of the method calls on builtin type constant could be resolved and const folded at compile time (example below), but currently they are being called at runtime.

![Capture](https://user-images.githubusercontent.com/41085900/91844090-75817000-ec74-11ea-9825-c6ee12121140.JPG)

**Comment by jkb0o** [comment]
I wonder if this alsco can ber resolved for static functions
```gdscript
const value = singleton.static_function()
```

**Comment by Calinou** [comment]
@jkb0o A static function doesn't necessarily have a constant value. For instance, it could return a random number.

**Comment by ThakeeNathees** [comment]
@jkb0o  It's possible to implement more semantic analysis in the future to check if the static function could be predicted at compile time and remove all the calls with the constant value. see: [constexpr in c++](https://docs.microsoft.com/en-us/cpp/cpp/constexpr-cpp?view=vs-2019#:~:text=constexpr%20indicates%20that%20the%20value,template%20arguments%20and%20array%20declarations.)

**Comment by jkb0o** [comment]
@Calinou yeah, that's not really a const. I actually miss some sort of class level variables initialized on compile/load time, something like `static func()`, but `static var` in GDScript sometimes, but it's a different issue.

**Comment by Calinou** [comment]
@jkb0o See https://github.com/godotengine/godot-proposals/issues/820.
